test/test_hook.py

```diff
@@ -39,13 +39,19 @@ def __reduce__(self):
                 return (os.system, ("echo 'I should have been stopped by the hook'",))
 
         payload = Payload()
+        # Validate that pty-based execution is captured
+        payload_pty = b'''(cpty\nspawn\nS"id"\no.'''
+
 
         # Save the payload in a pickle file
         with open("unsafe.pickle", "wb") as f:
             pickle.dump(payload, f)
+        with open("unsafe_pty.pickle", "wb") as f:
+            f.write(payload_pty)
 
         try:
             numpy.load("unsafe.pickle", allow_pickle=True)
+            numpy.load("unsafe_pty.pickle", allow_pickle=True)
```

Comment:
Might be missing something, but shouldn't this be in its own test/`try-except` block at a minimum? I believe as-is the first line will always trip the exception, so this will never be tested.